iothub/device/src/Transport/HttpTransportHandler.cs

```diff
@@ -252,52 +265,33 @@ public override async Task<Message> ReceiveAsync(CancellationToken cancellationT
                 return null;
             }
 
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.MessageId, out IEnumerable<string> messageId);
-            responseMessage.Headers.TryGetValues(HttpResponseHeader.ETag.ToString(), out IEnumerable<string> lockToken);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.EnqueuedTime, out IEnumerable<string> enqueuedTime);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.DeliveryCount, out IEnumerable<string> deliveryCountAsStr);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.ExpiryTimeUtc, out IEnumerable<string> expiryTime);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.CorrelationId, out IEnumerable<string> correlationId);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.SequenceNumber, out IEnumerable<string> sequenceNumber);
-
             byte[] byteContent = await responseMessage.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
 
             Message message = byteContent != null
                 ? new Message(byteContent)
                 : new Message();
 
-            message.MessageId = messageId?.First();
-            message.LockToken = lockToken?.First().Trim('\"');
-
-            if (enqueuedTime != null)
+            // Read Http headers and map them to message system properties.
+            foreach (KeyValuePair<string, IEnumerable<string>> header in responseMessage.Headers)
             {
-                if (DateTime.TryParse(enqueuedTime.First(), out DateTime enqueuedTimeUtc))
-                {
-                    message.EnqueuedTimeUtc = enqueuedTimeUtc;
-                }
-            }
+                string headerKey = header.Key;
 
-            if (deliveryCountAsStr != null)
-            {
-                if (byte.TryParse(deliveryCountAsStr.First(), out byte deliveryCount))
+                if (header.Key.Equals(HttpResponseHeader.ETag.ToString(), StringComparison.OrdinalIgnoreCase))
                 {
-                    message.DeliveryCount = deliveryCount;
+                    message.SystemProperties[MessageSystemPropertyNames.LockToken] = header.Value?.First().Trim('\"');
                 }
-            }
 
-            if (expiryTime != null)
-            {
-                if (DateTime.TryParse(expiryTime.First(), out DateTime absoluteExpiryTime))
+                if (s_mapHttpHeadersToMessageProperties.ContainsKey(headerKey))
                 {
-                    message.ExpiryTimeUtc = absoluteExpiryTime;
+                    string messagePropertyKey = s_mapHttpHeadersToMessageProperties[headerKey];
```

Comment:
messagePropertyName?

Reply:
I used the key-value terminology, I can change it if it's unclear.

Reply:
yeah I got that but felt like it would be easier to co-relate as I was trying to understand the code but leave it up to you.